codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/UnionGenerator.java

```diff
@@ -78,6 +89,42 @@ public void run() {
         }
     }
 
+    /**
+     * Generates union usage examples for documentation.
+     *
+     * @param writer the writer
+     */
+    public void generateUnionExamples(GoWriter writer) {
+        Symbol symbol = symbolProvider.toSymbol(shape);
+        Set<MemberShape> members = new TreeSet<>(shape.getAllMembers().values());
+
+        writer.openBlock("func Example$L_outputUsage() {", "}", symbol.getName(), () -> {
+            writer.write("var union $P", symbol);
+
+            writer.writeDocs("type switches can be used to check the union value");
+            writer.openBlock("switch v := union.(type) {", "}", () -> {
+                for (MemberShape member : members) {
+                    Symbol targetSymbol = symbolProvider.toSymbol(model.expectShape(member.getTarget()));
+                    Symbol memberSymbol = SymbolUtils.createValueSymbolBuilder(symbolProvider.toMemberName(member),
+                            symbol.getNamespace()).build();
+
+                    writer.openBlock("case *$T:", "", memberSymbol, () -> {
+                        writer.write("_ = v.Value // Value is $L", targetSymbol.getName());
```

Comment:
Instead of target symbol name, maybe this should be the type of the target symbol for value?

Reply:
Only difference between using $T vs getName is that $T probably will include the `types` package if needed?

Reply:
I believe this would have to modified some, as I believe the default out for symbol would be the full path including the `github.com/aws/aws-sdk-go-v2` portion of the path.